crates/extensions/tedge_config_manager/src/actor.rs

```diff
@@ -528,6 +567,92 @@ impl ConfigManagerActor {
     }
 }
 
+/// Writes a file atomically and optionally sets its permissions.
+///
+/// Setting permissions (owner and group) of a file is a privileged operation so it needs to be run
+/// as root. If the any of the filesystem operations fail due to not having permissions, the
+/// function will return an error.
```

Comment:
Addressed in 93c3eba.

tedge-agent will attempt to deploy configuration by copying directly, and then if it fails because of permission denied, will repeat the entire process using tedge-write. Because we're using `tempfile`, if the process fails tempfile will be deleted automatically. I needed to use tempfile directly to set permissions before the final rename. 

1. Tempfile with mode 600 but owned by tedge.
2. The file is already downloaded and resides in the temporary directory (`/tmp`), so this is only a copy, but we try to do it as tedge first.
3. As requested
4. As requested.

As for the `/etc/tedge/tedge.toml`, TODO in store function was fixed, so we can adjust default permissions of the tempfile in `atomically_write_file_*` functions in another PR.